src/metaswitch/clearwater/plugin_tests/test_queue_manager_plugin.py

```diff
@@ -0,0 +1,133 @@
+def fail_check_node_health(command):
+    if (command == "/usr/share/clearwater/clearwater-queue-manager/scripts/check_node_health.py"):
+        return False
+    else:
+        return True
+
+
+class TestQueueManagerPlugin(unittest.TestCase):
+    @mock.patch('clearwater_etcd_plugins.clearwater_queue_manager.apply_config_plugin.run_command')
+    @mock.patch('clearwater_etcd_plugins.clearwater_queue_manager.apply_config_plugin.os.path.exists')
+    @mock.patch('clearwater_etcd_plugins.clearwater_queue_manager.apply_config_plugin.os.listdir')
+    def test_front_of_queue(self, mock_os_listdir, mock_os_path_exists, mock_run_command):
+        """Test Queue Manager front_of_queue function"""
+
+        # Create the plugin
+        plugin = ApplyConfigPlugin\
+                    (PluginParams(ip='10.0.0.1',
+                                  mgmt_ip='10.0.1.1',
+                                  local_site='local_site',
+                                  remote_site='remote_site',
+                                  remote_cassandra_seeds='',
+                                  signaling_namespace='',
+                                  uuid=uuid.UUID('92a674aa-a64b-4549-b150-596fd466923f'),
+                                  etcd_key='etcd_key',
+                                  etcd_cluster_key='etcd_cluster_key'))
+
+        # Set up the mock environment and expectations
+        mock_os_path_exists.return_value = True
+        mock_os_listdir.return_value = ["test_restart_script"]
+
+        expected_command_call_list = \
+            [mock.call("service clearwater-infrastructure restart"),
+             mock.call("/usr/share/clearwater/infrastructure/scripts/restart/test_restart_script"),
+             mock.call("/usr/share/clearwater/clearwater-queue-manager/scripts/check_node_health.py"),
+             mock.call().__nonzero__(), # This call comes from the if statement test on the call above
+             mock.call("/usr/share/clearwater/clearwater-queue-manager/scripts/modify_nodes_in_queue"\
+                       " remove_failure apply_config")]
+
+        # Call the plugin hook
+        plugin.at_front_of_queue()
+
+        # Test our assertions
+        mock_os_path_exists.assert_called_once_with\
+                            ("/usr/share/clearwater/infrastructure/scripts/restart")
+        mock_os_listdir.assert_called_once_with\
+                            ("/usr/share/clearwater/infrastructure/scripts/restart")
+        mock_run_command.assert_has_calls(expected_command_call_list)
+
+
+    @mock.patch('clearwater_etcd_plugins.clearwater_queue_manager.apply_config_plugin.run_command',\
+                side_effect=fail_check_node_health)
+    @mock.patch('clearwater_etcd_plugins.clearwater_queue_manager.apply_config_plugin.os.path.exists')
+    @mock.patch('clearwater_etcd_plugins.clearwater_queue_manager.apply_config_plugin.os.listdir')
+    def test_front_of_queue_fail_node_health(self, mock_os_listdir,\
```

Comment:
I think you've got your test code the wrong way round - this test is for when check_node_health fails, but I think you've set it to succeed